### DISCUSSION TITLE
telemetry metrics

### DISCUSSION THREAD

**Discussion by BritneyStars** [apache/skywalking#10258] [Q&A] [answered]
In TraceSegmentHandler.java,the metric "trace_in_latency" define in 
  histogram = metricsCreator.createHistogramMetric(
            "trace_in_latency",
            "The process latency of trace data",
            new MetricsTag.Keys("protocol"),
            new MetricsTag.Values("kafka")
        );
but in oap.yml，the metric trace_in_latency change to trace_in_latency_count. 

- name: instance_trace_count
    exp: trace_in_latency_count.sum(['service', 'host_name']).increase('PT1M')

These are some metrics like this and end with _count,_percentile. I want to known where are these introductions.Thanks

**Comment by wu-sheng** [comment, marked as answer]
This is a general and widely used in the metrics system, such as Prometheus. You could learn Prometheus' histogram definition.

**Reply by BritneyStars** [reply]
Ok,Thank you very much

**Comment by yqjdcyy** [comment]
@wu-sheng how to generate the metric of trace_in_latency_count？without direct code, it maybe generate by MAL?

**Reply by wu-sheng** [reply]
What do you mean without direct code?

**Reply by yqjdcyy** [reply]
just like  `metricsCreator.createHistogramMetric("trace_in_latency", ...`

**Reply by wu-sheng** [reply]
Still not following. Please explain yourself clearly.

**Reply by yqjdcyy** [reply]
`trace_in_latency_count` metric work on grafana, i try to find the meaning of it, such as the count of segment or the count of batch.
but i cound not found where to generate the metric, without count code(`metricsCreator.createCounter`), neither config file(oap.yaml)

**Reply by wu-sheng** [reply]
If you mean why can't find `trace_in_latency_count` but only find `trace_in_latency` in the codes, that is Prometheus style, not ours.

Read https://prometheus.io/docs/practices/histograms/

**Reply by yqjdcyy** [reply]
I left out the part of prometheus's document myself，sorry and thx!

**Reply by yqjdcyy** [reply]
`PrometheusMetricConverter#convertMetric` seem to do for that
